### PR TITLE
Dispatcher logs

### PR DESCRIPTION
**Pull request by JakobMiksch**
- based on #138 
- Adds custom logging for the dispatcher to get an quick overview of what is happening on the server
- The logs are stored in a file in a JSON structure
- Logs are also send to standard out in an colorized and prettyfied way :heart_eyes: 
- The logs can be ready with CLI tools like `pino-pretty` as described in the README
- later they could be visualized with monitoring tools like Grafana, checkmk oder Kibana.
- Follow up: Use pino logging for all workers

### CODE REVIEW

File: src/dispatcher/dev.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  {},
+  pino.multistream(
+    [
+      pretty({ colorize: true }),
+      { stream: pino.destination(`./combined.log`) },
+    ]
+  )
+);
+
+logger.info('Asfdasdf')
```
Comment (hblitza) [inline src/dispatcher/dev.js:14]: ?

File: src/dispatcher/dispatcher.js [hunk 2]
```diff
@@ -46,139 +39,179 @@ let channel;
...
-    }
+  /**
+   * Creates a function that is executed when next task is available.
+   *
+   * Function needs to be created in this way to ensure it has access
+   * to the channel within its scope.
+   *
+   * @returns {Function} The function
+   */
+  #createHandleNextTaskFun() {
```
Comment (hblitza) [inline src/dispatcher/dispatcher.js:107]: #?

File: src/dispatcher/dev.js (hunk 1 above)
Comment (JakobMiksch) [reply src/dispatcher/dev.js]: good catch. I accidentally committed my dev script. I fixed it

File: src/dispatcher/dispatcher.js (hunk 2 above)
Comment (JakobMiksch) [reply src/dispatcher/dispatcher.js:107]: that's apparently the way to make functions private https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

**Comment by JakobMiksch** [comment]
closed in favour of #140

**Pull request closed without merging**